### DISCUSSION TITLE
Coil3 and bitmap configs

### DISCUSSION THREAD

**Discussion by saket** [coil-kt/coil#2641] [General]
Considering that coil3 uses `ImageDecoder` on android by default, is the `ImageRequest.Builder#bitmapConfig` option effectively no-op on API 29 and above?

**Comment by colinrtwhite** [comment]
It's still used to determine software vs. hardware, but good point we should probably disable decoding with `ImageDecoder` if bitmap config is set to `RGB_565` for instance.

As an aside, I was hoping to avoid these types of discrepancies by only using `BitmapFactory`, but we ended up using `ImageDecoder` as it [has slightly better IO performance](https://github.com/coil-kt/coil/issues/1943#issuecomment-1848878610). If we keep running into differences and it becomes tough to maintain it might make sense to reevaluate this.

cc @renovate-bot

**Reply by revonateB0T** [reply]
Maybe we could set `MEMORY_POLICY_LOW_RAM` for `RGB_565` 
https://developer.android.com/reference/android/graphics/ImageDecoder#MEMORY_POLICY_LOW_RAM
But seems it is not guaranteed.
Moreover it is not simply "slightly better IO performance", It allow us decode a `DirectByteBuffer` directly with https://github.com/coil-kt/coil/issues/1971 which is huge performance improvement.

**Reply by saket** [reply]
Switching to `BitmapFactory` for non-8888 config sounds like a good idea to me!
